s3/src/test/java/ch/cyberduck/core/cryptomator/UVFTest.java

```diff
@@ -0,0 +1,197 @@
+                final StatusOutputStream<?> out = storage.getFeature(Write.class).write(file, transferStatus, new DisabledConnectionCallback());
+                IOUtils.copyLarge(UVFTest.class.getResourceAsStream("/uvf/first_vault" + fi), out);
+                out.close();
+            }
+
+            final VaultRegistry vaults = new DefaultVaultRegistry(new DisabledPasswordCallback());
+            bookmark.setDefaultPath("/" + bucketName);
+            final UVFVault vault = new UVFVault(new DefaultPathHomeFeature(bookmark).find());
+            vaults.add(vault.load(storage, new DisabledPasswordCallback() {
+                @Override
+                public Credentials prompt(final Host bookmark, final String title, final String reason, final LoginOptions options) {
+                    return new Credentials().withPassword(jwe);
+                }
+            }));
+            final PathAttributes attr = storage.getFeature(AttributesFinder.class).find(vault.getHome());
+            storage.withRegistry(vaults);
+            try(final UVFMasterkey masterKey = UVFMasterkey.fromDecryptedPayload(jwe)) {
+                assertTrue(Arrays.equals(masterKey.rootDirId(), vault.getRootDirId()));
+            }
+
+            final Path home = vault.getHome().withAttributes(attr).withType(EnumSet.of(AbstractPath.Type.directory, AbstractPath.Type.vault));
+            {
+                final AttributedList<Path> list = storage.getFeature(ListService.class).list(home, new DisabledListProgressListener());
+                assertEquals(2, list.size());
+                assertTrue(Arrays.toString(list.toArray()), list.contains(new Path("/cyberduckbucket/foo.txt", EnumSet.of(AbstractPath.Type.file, AbstractPath.Type.decrypted))));
+                assertTrue(Arrays.toString(list.toArray()), list.contains(new Path("/cyberduckbucket/subdir", EnumSet.of(AbstractPath.Type.directory, AbstractPath.Type.placeholder, AbstractPath.Type.decrypted))));
+            }
+            {
+                final PathAttributes subdir = storage.getFeature(AttributesFinder.class).find(new Path("/cyberduckbucket/subdir", EnumSet.of(AbstractPath.Type.directory, AbstractPath.Type.placeholder, AbstractPath.Type.decrypted)));
+                // TODO still fails as test data from org.cryptomator.cryptolib.v3.UVFIntegrationTest uses latestSeed when creating dir.uvf, hard-coded in current implementation for subdir in DirectoryMetadata subDirMetadata = dirContentCryptor.newDirectoryMetadata();)
+                final AttributedList<Path> list = storage.getFeature(ListService.class).list(new Path("/cyberduckbucket/subdir", EnumSet.of(AbstractPath.Type.directory, AbstractPath.Type.placeholder, AbstractPath.Type.decrypted)).withAttributes(subdir), new DisabledListProgressListener());
+                assertEquals(1, list.size());
```

Comment:
@ylangisc with the `loadRevision` workaround, the encryption part of the subdir works and the raw folder is listed remotely, but then fails in `AbstractVault.decrypt` when attemptyting to decrypt the file name, probably due to the file name decryptor statically linked to the initial revision. Need to continue there.